x-pack/plugins/runtime_fields/public/components/runtime_field_form/runtime_field_form.tsx

```diff
@@ -0,0 +1,148 @@
+import { RUNTIME_FIELD_OPTIONS } from '../../constants';
+
+import { schema } from './schema';
+
+export interface FormState {
+  isValid: boolean | undefined;
+  isSubmitted: boolean;
+  submit: FormHook<RuntimeField>['submit'];
+}
+
+export interface Props {
+  docsBaseUri: string;
+  defaultValue?: RuntimeField;
+  onChange?: (state: FormState) => void;
+}
+
+const RuntimeFieldFormComp = ({ defaultValue, onChange, docsBaseUri }: Props) => {
+  const { form } = useForm<RuntimeField>({ defaultValue, schema });
+  const { submit, isValid: isFormValid, isSubmitted } = form;
+
+  useEffect(() => {
+    if (onChange) {
+      onChange({ isValid: isFormValid, isSubmitted, submit });
+    }
+  }, [onChange, isFormValid, isSubmitted, submit]);
+
+  return (
+    <Form form={form}>
+      <EuiFlexGroup>
+        {/* Name */}
+        <EuiFlexItem>
+          <UseField path="name" component={TextField} data-test-subj="nameField" />
+        </EuiFlexItem>
+
+        {/* Return type */}
+        <EuiFlexItem>
+          <UseField<EuiComboBoxOptionOption[]> path="type">
+            {({ label, value, setValue }) => {
+              if (value === undefined) {
+                return null;
+              }
+              return (
+                <>
+                  <EuiFormRow label={label} fullWidth>
+                    <EuiComboBox
+                      placeholder={i18n.translate(
+                        'xpack.runtimeFields.form.runtimeType.placeholderLabel',
+                        {
+                          defaultMessage: 'Select a type',
+                        }
+                      )}
+                      singleSelection={{ asPlainText: true }}
+                      options={RUNTIME_FIELD_OPTIONS}
+                      selectedOptions={value}
+                      onChange={(newValue) => {
+                        if (newValue.length === 0) {
+                          // Don't allow clearing the type. One must always be selected
+                          return;
+                        }
+                        setValue(newValue);
+                      }}
+                      isClearable={false}
+                      data-test-subj="typeField"
+                      fullWidth
+                    />
+                  </EuiFormRow>
+                </>
+              );
+            }}
+          </UseField>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiSpacer size="l" />
+
+      {/* Script */}
+      <UseField<string> path="script">
+        {({ value, setValue, label, helpText, isValid, getErrorsMessages }) => {
+          return (
+            <EuiFormRow
+              label={label}
+              error={getErrorsMessages()}
+              isInvalid={!isValid}
+              helpText={
+                <EuiFlexGroup justifyContent="flexEnd">
+                  <EuiFlexItem grow={false}>
+                    <EuiLink
+                      href={`${docsBaseUri}/to-be-defined`}
```

Comment:
I think I'd prefer to comment the link component out for now, rather than merge it with an invalid href.

Reply:
I have updated the href in my following PR to point correctly to https://www.elastic.co/guide/en/elasticsearch/painless/master/painless-lang-spec.html